psl-core/src/test/java/org/linqs/psl/application/inference/mpe/SGDOnlineInferenceTest.java

```diff
@@ -0,0 +1,179 @@
+package org.linqs.psl.application.inference.mpe;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.linqs.psl.TestModel;
+import org.linqs.psl.application.inference.InferenceApplication;
+import org.linqs.psl.application.inference.InferenceTest;
+import org.linqs.psl.database.Database;
+import org.linqs.psl.database.DatabaseTestUtil;
+import org.linqs.psl.database.atom.AtomManager;
+import org.linqs.psl.database.rdbms.driver.DatabaseDriver;
+import org.linqs.psl.model.atom.GroundAtom;
+import org.linqs.psl.model.atom.ObservedAtom;
+import org.linqs.psl.model.atom.QueryAtom;
+import org.linqs.psl.model.atom.RandomVariableAtom;
+import org.linqs.psl.model.formula.Conjunction;
+import org.linqs.psl.model.formula.Implication;
+import org.linqs.psl.model.predicate.GroundingOnlyPredicate;
+import org.linqs.psl.model.predicate.Predicate;
+import org.linqs.psl.model.predicate.StandardPredicate;
+import org.linqs.psl.model.rule.GroundRule;
+import org.linqs.psl.model.rule.Rule;
+import org.linqs.psl.model.rule.arithmetic.expression.coefficient.Add;
+import org.linqs.psl.model.rule.logical.WeightedGroundLogicalRule;
+import org.linqs.psl.model.rule.logical.WeightedLogicalRule;
+import org.linqs.psl.model.term.Constant;
+import org.linqs.psl.model.term.ConstantType;
+import org.linqs.psl.model.term.UniqueStringID;
+import org.linqs.psl.model.term.Variable;
+import org.linqs.psl.reasoner.function.FunctionTerm;
+import org.linqs.psl.reasoner.function.GeneralFunction;
+import org.linqs.psl.reasoner.sgd.term.SGDObjectiveTerm;
+import org.linqs.psl.reasoner.sgd.term.SGDStreamingTermStore;
+import org.linqs.psl.reasoner.sgd.term.SGDTermGenerator;
+import org.linqs.psl.reasoner.term.Hyperplane;
+import org.linqs.psl.reasoner.term.HyperplaneTermGenerator;
+import org.linqs.psl.server.actions.AddTerm;
+
+import java.util.*;
+
+public class SGDOnlineInferenceTest extends InferenceTest {
+    private TestModel.ModelInformation modelInfo;
+    private Database inferDB;
+    private SGDTermGenerator termGenerator;
+    private DatabaseDriver driver = DatabaseTestUtil.getPostgresDriver();
+    private Map<String, StandardPredicate> baselinePredicates = new HashMap<>();
+    private List<Rule> baselineRules = new ArrayList<Rule>();
+    private Map<StandardPredicate, List<TestModel.PredicateData>> baselineObservations = new HashMap<>();
+    private Map<StandardPredicate, List<TestModel.PredicateData>> baselineTargets = new HashMap<>();
+    private Map<StandardPredicate, List<TestModel.PredicateData>> baselineTruths = new HashMap<>();
+
+    /**
+     * Initialize a baseline model that we will be modifying with the online inference application
+     */
+    private void initBaselineModel() {
```

Comment:
(We discussed this in the SIG and I understand that you have just not gotten around to it yet, but I am including it for completeness.)

I think you should really consider just going with the default test model.
That model is in no way complete, and it cannot test every case.
However, using it is a lot simpler than having to make a full new model.
Are there any cases that you specifically need a new class for?